alternator/streams.cc

```diff
@@ -840,6 +840,9 @@ future<executor::request_return_type> executor::get_records(client_state& client
             {
                 auto item = rjson::empty_object();
                 describe_single_item(*selection, row, attr_names, item, true);
+                if (!item.ObjectEmpty() || op == cdc::operation::pre_image) {
```

Comment:
I don't understand, why do you need the "!item.ObjectEmpty()" part of this if? Didn't you just want to do this, unconditionally, for the pre_image?
Also, I think you maybe need to check || op == cdc::operation::post_image as well... Alternator's NEW_IMAGE also needs to include the key. As I mentioned above, the Alternator tests for this are missing (I'll do that) but I'm worried it will exhibit the same bug as the preimage.

Reply:
NEW_IMAGE will include it iff there is preimage data. I.e. if we did in fact to a delete, and no fields exist, there will be no NEW_IMAGE, neither keys nor fields. For OLD_IMAGE, we include it as long as there is one, because maybe keys-only.

Dropping postimage  for deleted rows would let us always include the keys, but as it is, the above two conditions (see the second `!item.ObjectEmpty()` below).

Reply:
I'm withdrawing my question about post_image, we can address these problem later when I write tests for them.
But I'm still puzzled about the "!item.ObjectEmpty()" thing. Apparently it's not needed for solving the pre_image-for-the-empty-but-existing-item bug, because the "op == pre_image" solves it. So what is it for?

Reply:
We add the keys if either:
  Any other value was included (basically NEW_IMAGE case, distinguishing row delete from something that has values)
or
  It is OLD_IMAGE, since pre_image is only generated iff there acually exists what we call OLD_IMAGE (previous state). 

Then there is a check below if `!item.ObjectEmpty()` to determine if we should in fact add OLD/NEW image at all, i.e. if we had either keys and/or data.

Reply:
Post image generation fix is included in the change set. Thus the checks can be removed.